### ISSUE TITLE
subprocess.CalledProcessError

### ISSUE THREAD

**Issue by CindyHuang23** [open-mmlab/mmdetection#1888]
Thanks for your error report and we appreciate it a lot.

**Checklist**
1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
I want to train my dataset on two gous with dist-train way, and to validate every epoch. But I got a error.

**Reproduction**
1. What command or script did you run?
```
(open-mmlab) lyy@lyy-pc:~/anaconda3/envs/open-mmlab/mmdetection/tools$ ./dist_train.sh '../configs/fcos/fcos_mstrain_640_800_x101_64x4d_fpn_gn_2x.py' 2 --validate

it is the command that I used to train my dataset
```
2. Did you make any modifications on the code or config? Did you understand what you have modified?

The code and config has been modified, and I have trained it successfully  on single gpu. It only  occur error when training on dist-train way.

3. What dataset did you use?
 DOTA1.0  dataset
**Environment**

1. The hardware environment: 2 * GeForce RTX 2080 Ti 12GB, CPU  31.1 GiB

2. The software environment : 
ubuntu 18.04

# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
addict                    2.2.1                     <pip>
blas                      1.1                    openblas    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
ca-certificates           2019.6.16            hecc5488_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
certifi                   2019.6.16                py36_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
cffi                      1.10.0                   py36_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
chardet                   3.0.4                     <pip>
cudatoolkit               10.0.130                      0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cython                    0.26                     py36_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
decorator                 4.4.1                     <pip>
freetype                  2.5.5                         2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
idna                      2.8                       <pip>
imageio                   2.6.1                     <pip>
imgaug                    0.3.0                     <pip>
intel-openmp              2019.4                      243    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
jbig                      2.1                           0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
jpeg                      9b                            0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
kiwisolver                1.1.0                     <pip>
libffi                    3.2.1                         1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
libgcc                    7.2.0                h69d50b8_2    defaults
libgcc-ng                 9.1.0                hdf63c60_0    defaults
libgfortran               3.0.0                         1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libpng                    1.6.30                        1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
libstdcxx-ng              9.1.0                hdf63c60_0    defaults
libtiff                   4.0.6                         3    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
libtorch                  0.1.12                  nomkl_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
matplotlib                3.1.1                     <pip>
mkl                       2019.4                      243    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
mmcv                      0.2.10                    <pip>
mmdet                     1.0rc0+8387aba            <pip>
ncurses                   6.1               hf484d3e_1002    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
networkx                  2.4                       <pip>
ninja                     1.9.0                h6bb024c_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
numpy                     1.13.3          py36_blas_openblas_200  [blas_openblas]  https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
numpy                     1.18.0                    <pip>
olefile                   0.44                     py36_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
openblas                  0.2.19                        2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
opencv-python             4.1.0.25                  <pip>
opencv-python-headless    4.1.2.30                  <pip>
openssl                   1.1.1c               h516909a_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
pandas                    0.25.3                    <pip>
Pillow                    6.2.1                     <pip>
pillow                    4.2.1                    py36_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
pip                       9.0.1                    py36_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
pip                       19.2.1                    <pip>
pycocotools               2.0.0                     <pip>
pycparser                 2.18                     py36_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
pyparsing                 2.4.2a1                   <pip>
python                    3.6.7             h357f687_1005    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
pytorch                   1.1.0           py3.6_cuda10.0.130_cudnn7.5.1_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
pytz                      2019.3                    <pip>
PyWavelets                1.1.1                     <pip>
PyYAML                    5.1.1                     <pip>
readline                  8.0                  hf8c457e_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
requests                  2.22.0                    <pip>
scikit-image              0.16.2                    <pip>
scipy                     1.4.1                     <pip>
seaborn                   0.9.0                     <pip>
setuptools                36.4.0                   py36_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
Shapely                   1.6.4.post2               <pip>
six                       1.10.0                   py36_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
sqlite                    3.29.0               hcee41ef_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
tk                        8.6.9             hed695b0_1002    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
torchvision               0.3.0           py36_cu10.0.130_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch
tqdm                      4.39.0                    <pip>
urllib3                   1.25.3                    <pip>
wheel                     0.29.0                   py36_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free
xz                        5.2.4             h14c3975_1001    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
zlib                      1.2.11                        0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free

**Error traceback**
If applicable, paste the error trackback here.
```
True
自动调整学习率
True
自动调整学习率
2019-12-28 15:52:00,577 - INFO - Distributed training: True
2019-12-28 15:52:01,209 - INFO - load model from: modelzoo://resnet101
/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/mmcv/runner/checkpoint.py:141: UserWarning: The URL scheme of "modelzoo://" is deprecated, please use "torchvision://" instead
  warnings.warn('The URL scheme of "modelzoo://" is deprecated, please '
/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/mmcv/runner/checkpoint.py:141: UserWarning: The URL scheme of "modelzoo://" is deprecated, please use "torchvision://" instead
  warnings.warn('The URL scheme of "modelzoo://" is deprecated, please '
2019-12-28 15:52:06,237 - WARNING - unexpected key in source state_dict: fc.weight, fc.bias

missing keys in source state_dict: layer3.21.bn1.num_batches_tracked, layer4.1.bn3.num_batches_tracked, layer3.20.bn3.num_batches_tracked, layer3.9.bn2.num_batches_tracked, layer4.0.bn1.num_batches_tracked, layer2.3.bn1.num_batches_tracked, layer3.10.bn1.num_batches_tracked, layer2.1.bn1.num_batches_tracked, layer3.21.bn2.num_batches_tracked, layer3.0.downsample.1.num_batches_tracked, layer1.1.bn2.num_batches_tracked, layer2.3.bn3.num_batches_tracked, layer3.18.bn2.num_batches_tracked, layer4.2.bn1.num_batches_tracked, layer3.15.bn2.num_batches_tracked, layer3.8.bn1.num_batches_tracked, layer2.3.bn2.num_batches_tracked, layer4.1.bn2.num_batches_tracked, layer3.14.bn3.num_batches_tracked, layer3.7.bn2.num_batches_tracked, layer3.4.bn3.num_batches_tracked, layer3.18.bn3.num_batches_tracked, layer3.11.bn2.num_batches_tracked, layer3.19.bn2.num_batches_tracked, layer1.2.bn1.num_batches_tracked, layer3.11.bn3.num_batches_tracked, layer3.9.bn1.num_batches_tracked, layer1.2.bn2.num_batches_tracked, layer3.3.bn2.num_batches_tracked, layer2.2.bn3.num_batches_tracked, layer1.0.bn2.num_batches_tracked, layer1.2.bn3.num_batches_tracked, layer1.1.bn1.num_batches_tracked, layer3.8.bn2.num_batches_tracked, layer3.12.bn3.num_batches_tracked, layer3.8.bn3.num_batches_tracked, layer3.5.bn1.num_batches_tracked, layer3.12.bn1.num_batches_tracked, layer3.18.bn1.num_batches_tracked, layer3.11.bn1.num_batches_tracked, layer3.12.bn2.num_batches_tracked, layer2.2.bn1.num_batches_tracked, layer3.22.bn3.num_batches_tracked, layer3.19.bn3.num_batches_tracked, layer4.0.bn2.num_batches_tracked, layer3.13.bn1.num_batches_tracked, layer3.4.bn2.num_batches_tracked, layer3.1.bn3.num_batches_tracked, layer3.16.bn3.num_batches_tracked, layer3.20.bn2.num_batches_tracked, layer3.6.bn2.num_batches_tracked, layer2.0.bn1.num_batches_tracked, layer3.14.bn2.num_batches_tracked, layer3.17.bn3.num_batches_tracked, layer3.2.bn3.num_batches_tracked, layer3.0.bn1.num_batches_tracked, layer3.2.bn1.num_batches_tracked, layer2.2.bn2.num_batches_tracked, layer3.5.bn3.num_batches_tracked, layer3.22.bn1.num_batches_tracked, layer2.0.bn3.num_batches_tracked, layer3.4.bn1.num_batches_tracked, layer3.7.bn1.num_batches_tracked, layer3.15.bn3.num_batches_tracked, layer3.3.bn1.num_batches_tracked, layer2.1.bn2.num_batches_tracked, bn1.num_batches_tracked, layer3.17.bn2.num_batches_tracked, layer4.0.bn3.num_batches_tracked, layer3.9.bn3.num_batches_tracked, layer4.0.downsample.1.num_batches_tracked, layer2.0.downsample.1.num_batches_tracked, layer3.13.bn2.num_batches_tracked, layer3.10.bn3.num_batches_tracked, layer3.5.bn2.num_batches_tracked, layer3.17.bn1.num_batches_tracked, layer3.6.bn3.num_batches_tracked, layer3.21.bn3.num_batches_tracked, layer3.6.bn1.num_batches_tracked, layer3.0.bn3.num_batches_tracked, layer3.14.bn1.num_batches_tracked, layer3.15.bn1.num_batches_tracked, layer2.1.bn3.num_batches_tracked, layer3.7.bn3.num_batches_tracked, layer1.1.bn3.num_batches_tracked, layer3.20.bn1.num_batches_tracked, layer3.1.bn2.num_batches_tracked, layer1.0.bn1.num_batches_tracked, layer3.22.bn2.num_batches_tracked, layer3.16.bn1.num_batches_tracked, layer3.19.bn1.num_batches_tracked, layer4.1.bn1.num_batches_tracked, layer3.1.bn1.num_batches_tracked, layer4.2.bn2.num_batches_tracked, layer3.3.bn3.num_batches_tracked, layer4.2.bn3.num_batches_tracked, layer3.10.bn2.num_batches_tracked, layer3.13.bn3.num_batches_tracked, layer3.16.bn2.num_batches_tracked, layer3.2.bn2.num_batches_tracked, layer2.0.bn2.num_batches_tracked, layer3.0.bn2.num_batches_tracked, layer1.0.bn3.num_batches_tracked, layer1.0.downsample.1.num_batches_tracked

loading annotations into memory...
loading annotations into memory...
Done (t=1.51s)
creating index...
Done (t=1.31s)
creating index...
index created!
index created!
loading annotations into memory...
loading annotations into memory...
Done (t=0.60s)
creating index...
Done (t=0.60s)
creating index...
index created!
index created!
2019-12-28 15:52:08,763 - INFO - Start running, host: lyy@lyy-pc, work_dir: /home/lyy/anaconda3/envs/open-mmlab/mmdetection/tools/work_dirs/PX/DOTA/val+fcos_mstrain_640_800_X101_dpafpn+giou(lw2)+creg+ctr_sample_gn(1.5)_2x
2019-12-28 15:52:08,764 - INFO - workflow: [('train', 1)], max: 24 epochs
2019-12-28 15:52:30,875 - INFO - Epoch [1][50/2569]	lr: 0.00042, eta: 7:33:58, time: 0.442, data_time: 0.108, memory: 4047, loss_cls: 0.8181, loss_bbox: 1.7639, loss_centerness: 0.7000, loss: 3.2821
2019-12-28 15:52:47,790 - INFO - Epoch [1][100/2569]	lr: 0.00042, eta: 6:40:19, time: 0.338, data_time: 0.003, memory: 4071, loss_cls: 0.7565, loss_bbox: 1.3699, loss_centerness: 0.6909, loss: 2.8172
2019-12-28 15:53:04,518 - INFO - Epoch [1][150/2569]	lr: 0.00042, eta: 6:20:59, time: 0.335, data_time: 0.003, memory: 4198, loss_cls: 0.7062, loss_bbox: 1.2846, loss_centerness: 0.6841, loss: 2.6750
2019-12-28 15:53:21,379 - INFO - Epoch [1][200/2569]	lr: 0.00042, eta: 6:11:51, time: 0.337, data_time: 0.003, memory: 4198, loss_cls: 0.6821, loss_bbox: 1.2539, loss_centerness: 0.6773, loss: 2.6134
2019-12-28 15:53:38,145 - INFO - Epoch [1][250/2569]	lr: 0.00042, eta: 6:05:52, time: 0.335, data_time: 0.003, memory: 4293, loss_cls: 0.6100, loss_bbox: 1.2322, loss_centerness: 0.6727, loss: 2.5149
2019-12-28 15:53:54,929 - INFO - Epoch [1][300/2569]	lr: 0.00042, eta: 6:01:51, time: 0.336, data_time: 0.003, memory: 4293, loss_cls: 0.5308, loss_bbox: 1.1281, loss_centerness: 0.6728, loss: 2.3316
2019-12-28 15:54:11,733 - INFO - Epoch [1][350/2569]	lr: 0.00042, eta: 5:58:58, time: 0.336, data_time: 0.004, memory: 4293, loss_cls: 0.5485, loss_bbox: 1.0870, loss_centerness: 0.6704, loss: 2.3059
2019-12-28 15:54:28,331 - INFO - Epoch [1][400/2569]	lr: 0.00042, eta: 5:56:12, time: 0.332, data_time: 0.003, memory: 4293, loss_cls: 0.5275, loss_bbox: 1.0684, loss_centerness: 0.6748, loss: 2.2708
2019-12-28 15:54:45,041 - INFO - Epoch [1][450/2569]	lr: 0.00042, eta: 5:54:14, time: 0.334, data_time: 0.004, memory: 4323, loss_cls: 0.5765, loss_bbox: 1.1212, loss_centerness: 0.6692, loss: 2.3670
2019-12-28 15:55:01,760 - INFO - Epoch [1][500/2569]	lr: 0.00042, eta: 5:52:38, time: 0.334, data_time: 0.004, memory: 4323, loss_cls: 0.4974, loss_bbox: 1.0941, loss_centerness: 0.6743, loss: 2.2658
2019-12-28 15:55:18,982 - INFO - Epoch [1][550/2569]	lr: 0.00125, eta: 5:52:12, time: 0.344, data_time: 0.003, memory: 4323, loss_cls: 0.6971, loss_bbox: 1.2325, loss_centerness: 0.6768, loss: 2.6064
2019-12-28 15:55:36,221 - INFO - Epoch [1][600/2569]	lr: 0.00125, eta: 5:51:49, time: 0.345, data_time: 0.003, memory: 4323, loss_cls: 0.6831, loss_bbox: 1.1145, loss_centerness: 0.6815, loss: 2.4791
2019-12-28 15:55:53,410 - INFO - Epoch [1][650/2569]	lr: 0.00125, eta: 5:51:23, time: 0.344, data_time: 0.003, memory: 4323, loss_cls: 0.5893, loss_bbox: 1.1437, loss_centerness: 0.6766, loss: 2.4095
2019-12-28 15:56:10,598 - INFO - Epoch [1][700/2569]	lr: 0.00125, eta: 5:50:58, time: 0.344, data_time: 0.004, memory: 4323, loss_cls: 0.6788, loss_bbox: 1.1277, loss_centerness: 0.6785, loss: 2.4850
2019-12-28 15:56:27,246 - INFO - Epoch [1][750/2569]	lr: 0.00125, eta: 5:49:49, time: 0.333, data_time: 0.003, memory: 4323, loss_cls: 0.5892, loss_bbox: 1.1021, loss_centerness: 0.6802, loss: 2.3714
2019-12-28 15:56:44,350 - INFO - Epoch [1][800/2569]	lr: 0.00125, eta: 5:49:23, time: 0.342, data_time: 0.003, memory: 4323, loss_cls: 0.5019, loss_bbox: 1.1099, loss_centerness: 0.6745, loss: 2.2864
2019-12-28 15:57:01,894 - INFO - Epoch [1][850/2569]	lr: 0.00125, eta: 5:49:28, time: 0.351, data_time: 0.003, memory: 4323, loss_cls: 0.4548, loss_bbox: 0.9895, loss_centerness: 0.6664, loss: 2.1107
2019-12-28 15:57:19,282 - INFO - Epoch [1][900/2569]	lr: 0.00125, eta: 5:49:21, time: 0.348, data_time: 0.004, memory: 4323, loss_cls: 0.4745, loss_bbox: 1.0320, loss_centerness: 0.6775, loss: 2.1840
2019-12-28 15:57:35,905 - INFO - Epoch [1][950/2569]	lr: 0.00125, eta: 5:48:23, time: 0.332, data_time: 0.003, memory: 4323, loss_cls: 0.5412, loss_bbox: 0.9371, loss_centerness: 0.6647, loss: 2.1430
2019-12-28 15:57:53,093 - INFO - Epoch [1][1000/2569]	lr: 0.00125, eta: 5:48:04, time: 0.344, data_time: 0.004, memory: 4323, loss_cls: 0.5324, loss_bbox: 0.9603, loss_centerness: 0.6658, loss: 2.1585
2019-12-28 15:58:10,737 - INFO - Epoch [1][1050/2569]	lr: 0.00125, eta: 5:48:12, time: 0.353, data_time: 0.004, memory: 4323, loss_cls: 0.4535, loss_bbox: 0.9440, loss_centerness: 0.6586, loss: 2.0561
2019-12-28 15:58:28,539 - INFO - Epoch [1][1100/2569]	lr: 0.00125, eta: 5:48:26, time: 0.356, data_time: 0.004, memory: 4323, loss_cls: 0.4518, loss_bbox: 0.9427, loss_centerness: 0.6659, loss: 2.0604
2019-12-28 15:58:46,149 - INFO - Epoch [1][1150/2569]	lr: 0.00125, eta: 5:48:27, time: 0.352, data_time: 0.003, memory: 4323, loss_cls: 0.4663, loss_bbox: 0.9354, loss_centerness: 0.6590, loss: 2.0607
2019-12-28 15:59:03,685 - INFO - Epoch [1][1200/2569]	lr: 0.00125, eta: 5:48:23, time: 0.351, data_time: 0.003, memory: 4323, loss_cls: 0.4430, loss_bbox: 0.9439, loss_centerness: 0.6616, loss: 2.0485
2019-12-28 15:59:21,707 - INFO - Epoch [1][1250/2569]	lr: 0.00125, eta: 5:48:41, time: 0.360, data_time: 0.004, memory: 4323, loss_cls: 0.4936, loss_bbox: 0.9998, loss_centerness: 0.6730, loss: 2.1663
2019-12-28 15:59:39,489 - INFO - Epoch [1][1300/2569]	lr: 0.00125, eta: 5:48:45, time: 0.356, data_time: 0.004, memory: 4323, loss_cls: 0.5080, loss_bbox: 0.9797, loss_centerness: 0.6702, loss: 2.1578
2019-12-28 15:59:57,202 - INFO - Epoch [1][1350/2569]	lr: 0.00125, eta: 5:48:45, time: 0.354, data_time: 0.003, memory: 4323, loss_cls: 0.4738, loss_bbox: 0.9074, loss_centerness: 0.6613, loss: 2.0425
2019-12-28 16:00:14,813 - INFO - Epoch [1][1400/2569]	lr: 0.00125, eta: 5:48:38, time: 0.352, data_time: 0.004, memory: 4323, loss_cls: 0.4059, loss_bbox: 0.8823, loss_centerness: 0.6678, loss: 1.9560
2019-12-28 16:00:32,475 - INFO - Epoch [1][1450/2569]	lr: 0.00125, eta: 5:48:34, time: 0.353, data_time: 0.003, memory: 4323, loss_cls: 0.3777, loss_bbox: 0.9247, loss_centerness: 0.6639, loss: 1.9663
2019-12-28 16:00:50,009 - INFO - Epoch [1][1500/2569]	lr: 0.00125, eta: 5:48:23, time: 0.351, data_time: 0.004, memory: 4323, loss_cls: 0.4421, loss_bbox: 0.9830, loss_centerness: 0.6603, loss: 2.0854
2019-12-28 16:01:07,560 - INFO - Epoch [1][1550/2569]	lr: 0.00125, eta: 5:48:12, time: 0.351, data_time: 0.003, memory: 4323, loss_cls: 0.3718, loss_bbox: 0.8846, loss_centerness: 0.6640, loss: 1.9203
2019-12-28 16:01:25,127 - INFO - Epoch [1][1600/2569]	lr: 0.00125, eta: 5:48:02, time: 0.351, data_time: 0.003, memory: 4323, loss_cls: 0.3773, loss_bbox: 0.8242, loss_centerness: 0.6620, loss: 1.8635
2019-12-28 16:01:43,022 - INFO - Epoch [1][1650/2569]	lr: 0.00125, eta: 5:48:03, time: 0.358, data_time: 0.003, memory: 4560, loss_cls: 0.4264, loss_bbox: 0.8353, loss_centerness: 0.6578, loss: 1.9194
2019-12-28 16:02:00,757 - INFO - Epoch [1][1700/2569]	lr: 0.00125, eta: 5:47:57, time: 0.355, data_time: 0.003, memory: 4560, loss_cls: 0.4372, loss_bbox: 0.8755, loss_centerness: 0.6616, loss: 1.9743
2019-12-28 16:02:18,150 - INFO - Epoch [1][1750/2569]	lr: 0.00125, eta: 5:47:39, time: 0.348, data_time: 0.004, memory: 4560, loss_cls: 0.3899, loss_bbox: 0.9216, loss_centerness: 0.6553, loss: 1.9667
2019-12-28 16:02:36,263 - INFO - Epoch [1][1800/2569]	lr: 0.00125, eta: 5:47:45, time: 0.362, data_time: 0.003, memory: 4560, loss_cls: 0.3707, loss_bbox: 0.8170, loss_centerness: 0.6613, loss: 1.8490
2019-12-28 16:02:54,353 - INFO - Epoch [1][1850/2569]	lr: 0.00125, eta: 5:47:49, time: 0.362, data_time: 0.004, memory: 4560, loss_cls: 0.3609, loss_bbox: 0.8769, loss_centerness: 0.6592, loss: 1.8970
2019-12-28 16:03:12,306 - INFO - Epoch [1][1900/2569]	lr: 0.00125, eta: 5:47:48, time: 0.359, data_time: 0.004, memory: 4560, loss_cls: 0.3621, loss_bbox: 0.8026, loss_centerness: 0.6530, loss: 1.8177
2019-12-28 16:03:30,219 - INFO - Epoch [1][1950/2569]	lr: 0.00125, eta: 5:47:44, time: 0.358, data_time: 0.004, memory: 4560, loss_cls: 0.4064, loss_bbox: 0.8351, loss_centerness: 0.6592, loss: 1.9007
2019-12-28 16:03:48,161 - INFO - Epoch [1][2000/2569]	lr: 0.00125, eta: 5:47:41, time: 0.359, data_time: 0.003, memory: 4560, loss_cls: 0.3659, loss_bbox: 0.8265, loss_centerness: 0.6522, loss: 1.8446
2019-12-28 16:04:06,318 - INFO - Epoch [1][2050/2569]	lr: 0.00125, eta: 5:47:43, time: 0.363, data_time: 0.003, memory: 4560, loss_cls: 0.4201, loss_bbox: 0.8610, loss_centerness: 0.6577, loss: 1.9389
2019-12-28 16:04:24,091 - INFO - Epoch [1][2100/2569]	lr: 0.00125, eta: 5:47:33, time: 0.355, data_time: 0.003, memory: 4560, loss_cls: 0.3577, loss_bbox: 0.8727, loss_centerness: 0.6488, loss: 1.8792
2019-12-28 16:04:42,179 - INFO - Epoch [1][2150/2569]	lr: 0.00125, eta: 5:47:31, time: 0.362, data_time: 0.003, memory: 4560, loss_cls: 0.3748, loss_bbox: 0.7902, loss_centerness: 0.6498, loss: 1.8148
2019-12-28 16:05:00,517 - INFO - Epoch [1][2200/2569]	lr: 0.00125, eta: 5:47:36, time: 0.367, data_time: 0.003, memory: 4560, loss_cls: 0.3618, loss_bbox: 0.7506, loss_centerness: 0.6526, loss: 1.7650
2019-12-28 16:05:18,742 - INFO - Epoch [1][2250/2569]	lr: 0.00125, eta: 5:47:36, time: 0.364, data_time: 0.003, memory: 4560, loss_cls: 0.3423, loss_bbox: 0.8241, loss_centerness: 0.6488, loss: 1.8152
2019-12-28 16:05:37,198 - INFO - Epoch [1][2300/2569]	lr: 0.00125, eta: 5:47:42, time: 0.369, data_time: 0.003, memory: 4560, loss_cls: 0.3359, loss_bbox: 0.8338, loss_centerness: 0.6515, loss: 1.8212
2019-12-28 16:05:55,329 - INFO - Epoch [1][2350/2569]	lr: 0.00125, eta: 5:47:39, time: 0.363, data_time: 0.003, memory: 4560, loss_cls: 0.3444, loss_bbox: 0.7828, loss_centerness: 0.6525, loss: 1.7797
2019-12-28 16:06:14,017 - INFO - Epoch [1][2400/2569]	lr: 0.00125, eta: 5:47:48, time: 0.374, data_time: 0.004, memory: 4560, loss_cls: 0.3310, loss_bbox: 0.7851, loss_centerness: 0.6512, loss: 1.7674
2019-12-28 16:06:32,569 - INFO - Epoch [1][2450/2569]	lr: 0.00125, eta: 5:47:53, time: 0.371, data_time: 0.003, memory: 4560, loss_cls: 0.3398, loss_bbox: 0.7488, loss_centerness: 0.6504, loss: 1.7389
2019-12-28 16:06:50,952 - INFO - Epoch [1][2500/2569]	lr: 0.00125, eta: 5:47:54, time: 0.368, data_time: 0.003, memory: 4560, loss_cls: 0.3836, loss_bbox: 0.8060, loss_centerness: 0.6475, loss: 1.8372
2019-12-28 16:07:09,960 - INFO - Epoch [1][2550/2569]	lr: 0.00125, eta: 5:48:08, time: 0.380, data_time: 0.003, memory: 4560, loss_cls: 0.3220, loss_bbox: 0.7499, loss_centerness: 0.6456, loss: 1.7175
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 3350/3349, 17.5 task/s, elapsed: 192s, ETA:     0s

Loading and preparing results...
DONE (t=1.83s)
creating index...
index created!
Traceback (most recent call last):
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/numpy/core/function_base.py", line 117, in linspace
    num = operator.index(num)
TypeError: 'numpy.float64' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./train.py", line 108, in <module>
    main()
  File "./train.py", line 104, in main
    logger=logger)
  File "/home/lyy/hq/congqing/mmdet/apis/train.py", line 60, in train_detector
    _dist_train(model, dataset, cfg, validate=validate)
  File "/home/lyy/hq/congqing/mmdet/apis/train.py", line 189, in _dist_train
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs)
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/mmcv/runner/runner.py", line 358, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/mmcv/runner/runner.py", line 274, in train
    self.call_hook('after_train_epoch')
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/mmcv/runner/runner.py", line 229, in call_hook
    getattr(hook, fn_name)(self)
  File "/home/lyy/hq/congqing/mmdet/core/evaluation/eval_hooks.py", line 64, in after_train_epoch
    self.evaluate(runner, results)
  File "/home/lyy/hq/congqing/mmdet/core/evaluation/eval_hooks.py", line 153, in evaluate
    cocoEval = COCOeval(cocoGt, cocoDt, iou_type)
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/pycocotools-2.0.0-py3.6-linux-x86_64.egg/pycocotools/cocoeval.py", line 76, in __init__
    self.params = Params(iouType=iouType) # parameters
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/pycocotools-2.0.0-py3.6-linux-x86_64.egg/pycocotools/cocoeval.py", line 527, in __init__
    self.setDetParams()
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/pycocotools-2.0.0-py3.6-linux-x86_64.egg/pycocotools/cocoeval.py", line 507, in setDetParams
    self.iouThrs = np.linspace(.5, 0.95, np.round((0.95 - .5) / .05) + 1, endpoint=True)
  File "<__array_function__ internals>", line 6, in linspace
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/numpy/core/function_base.py", line 121, in linspace
    .format(type(num)))
TypeError: object of type <class 'numpy.float64'> cannot be safely interpreted as an integer.
Traceback (most recent call last):
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/torch/distributed/launch.py", line 235, in <module>
    main()
  File "/home/lyy/anaconda3/envs/open-mmlab/lib/python3.6/site-packages/torch/distributed/launch.py", line 231, in main
    cmd=process.args)
subprocess.CalledProcessError: Command '['/home/lyy/anaconda3/envs/open-mmlab/bin/python', '-u', './train.py', '--local_rank=0', '--config=../configs/fcos/fcos_mstrain_640_800_x101_64x4d_fpn_gn_2x.py', '--launcher', 'pytorch']' returned non-zero exit status 1.

```

Thanks  a lot.

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by changqinyao** [comment]
I have the same problem

**Comment by hellock** [comment]
The key error message is `TypeError: 'numpy.float64' object cannot be interpreted as an integer`. You may first identify which index is of type `numpy.float64` instead of int.

**Comment by thangvubk** [comment]
This is due to numpy 1.18+ I guess (refer to [here](https://github.com/cocodataset/cocoapi/issues/356)). A workaround is downgrade numpy to 1.17 or wait for the PR merge.

**Comment by hellock** [comment]
Fixed in #2024.
